regex-syntax/src/lib.rs

```diff
@@ -105,6 +105,21 @@ done automatically in the `regex` crate.
 
 #![deny(missing_docs)]
 
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(all(feature = "alloc", not(feature = "std")), feature(alloc))]
+#![cfg_attr(all(feature = "alloc", not(feature = "std")), feature(slice_concat_ext))]
+
+#[cfg(test)]
+extern crate std as std_test;
+
+#[cfg(feature = "std")]
+#[macro_use]
+extern crate std as core;
```

Comment:
In retrospect, I would recommend not doing what we did in Rand but instead having `extern crate core as std;` (for non-std). The reasons being (a) it saves changing so many imports and (b) `std` may use feature flags in the future.

However, I suppose error messages may be fun in no-std mode :smile:

Reply:
I adopted this recommendation in the latest commit to this PR.